src/lib/url.js

```diff
@@ -0,0 +1,64 @@
+const queryParams = ['include', 'fields', 'sort'];
+
+export const parseInclude = include => {
+  return include ? include.split(',') : [];
+};
+
+export const parseFields = query => {
+  let fields = {};
+
+  if (query) {
+    for (let [key, value] of query) {
+      if (key.startsWith('fields') && key.indexOf('[') > -1) {
+        const type = key.substring(key.indexOf('[') + 1, key.indexOf(']'));
+        fields[type] = new Set(value.split(','));
```

Comment:
I like these two lines. It would be tempting to use a regex, but I think the more mechanical index and substring based approach is appropriate because this syntax is defined by the spec.

Reply:
Do you think we will need a better way to "process" out these keys- specifically with the more nested syntax of filters `filter[author-filter][condition][path]=uid.id` -here it works but may need to be more robust.

Reply:
What do you mean by a better way? We'll definitely need to modify this or write something else to handle the deeper "nesting". There might be an npm package, but it also wouldn't be terribly difficult to write ourselves. Could be fun to pair on actually.